kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/Tls.java

```diff
@@ -8,17 +8,40 @@
 
 import java.security.KeyStore;
 import java.util.Locale;
+import java.util.Objects;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
 
 /**
  * Provides TLS configuration for this peer.  This class is designed to be used for both TLS server and client roles.
  *
  * @param key   specifies a key provider that provides the certificate/key used to identify this peer.
  * @param trust specifies a trust provider used by this peer to determine whether to trust the peer. If omitted platform trust is used instead.
+ * @param clientAuth specifies how the TLS session should authenticate clients. Optional not required when configuring upstream TLS
```

Comment:
```suggestion
 * @param clientAuth when in the node is in the TLS <em>server</em> role, clientAuth specifies how the server should authenticate clients. This option has no meaning in the node is in the TLS <em>client</em> role. 
```

We've used the expression TLS server and TLS client elsewhere in the Javadoc to describe how options apply to the different roles.  I think it make sense to adopt the same language here.